### PR TITLE
Add README, refactor E2E tests

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Just some cleanup before a release

### Changes

- Move E2E tests to `e2e/bitrise.yaml`
- Generate new README based on `step.yml`

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -1,142 +1,17 @@
...
 
-app:
-  envs:
-    - SAMPLE_APP_URL: https://github.com/bitrise-samples/sample-apps-flutter-ios-android.git
-    - SAMPLE_APP_BRANCH: master
-    - ORIGIN_SOURCE_DIR: $BITRISE_SOURCE_DIR
-    - PROJECT_LOCATION: "$BITRISE_SOURCE_DIR/_tmp"
-    - GENERATE_CODE_COVERAGE_FILES: "no"
-
 workflows:
```
Comment (Bence1001) [inline bitrise.yml:4]: I think we should add a `sample` workflow if we can.

File: bitrise.yml [hunk 2]
```diff
@@ -1,142 +1,17 @@
...
-  audit-this-step:
+  generate_readme:
     steps:
-      - script:
-          inputs:
-            - content: |-
-                #!/bin/bash
-                set -ex
-                stepman audit --step-yml ./step.yml
+    - git::https://github.com/bitrise-steplib/steps-readme-generator.git@main: {}
```
Comment (Bence1001) [inline bitrise.yml:17]: `docs/examples.md` is not used here (it's also an empty file). Do we need it?

File: step.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-title: "Flutter Test"
```
Comment (Bence1001) [inline step.yml:1]: Just out of curiosity: why did we remove the quotations in this file? I know YAML doesn't necessarily need quotes for strings, but we use them in almost all `step.yml`'s I've seen.

**Review by Bence1001 — COMMENTED** [review]
I've got a couple of comments but other than that it looks good to me.

File: step.yml (hunk 3 above)
Comment (ofalvai) [reply step.yml:1]: I like the quotation-less version, so I remove it wherever it makes sense. We don't have a YAML style guide (yet)

Comment (Bence1001) [reply step.yml:1]: Although we don't at the moment, I think we should strive for some level of consistency. Therefore, I'd suggest not removing these now. Wdyt?

**Review by Bence1001 — APPROVED** [review]

**Pull request merged**